ctapipe/io/metadata.py

```diff
@@ -0,0 +1,222 @@
+"""
+Management of CTA Reference Metadata, as defined in the CTA Top-Level Data Model
+document [ctatopleveldatamodel]_ , version 1A. This information is required to be
+attached to the header of any
+files generated.
+
+The class Reference collects all required reference metadata, and can be turned into a
+flat dictionary.  The user should try to fill out all fields, or use a helper to fill
+them (as in `Activity.from_provenance()`)
+
+.. code-block:: python
+
+    ref = Reference(
+        contact=Contact(name="Some User", email="user@me.com"),
+        product=Product(format='hdf5', ...),
+        process=Process(...),
+        activity=Activity(...),
+        instrument = Instrument(...)
+    )
+
+    some_astropy_table.meta = ref.to_dict()
+    some_astropy_table.write("output.ecsv")
+
+"""
+import uuid
+from collections import OrderedDict
+import warnings
+
+from astropy.time import Time
+from traitlets import (
+    Enum,
+    Unicode,
+    Int,
+    HasTraits,
+    default,
+    Instance,
+    validate,
+)
+
+from ctapipe.core.provenance import _ActivityProvenance
+
+__all__ = [
+    "Reference",
+    "Contact",
+    "Process",
+    "Product",
+    "Activity",
+    "Instrument",
+    "write_to_hdf5",
+]
+
+
+class Contact(HasTraits):
```

Comment:
Don't these all have to be config=True so one can set it in a config file?

Reply:
these aren't configuration options, I'm use using traitlets to do the validation.

Reply:
Why shouldn't the analyser be able to set his or her info in the config?

Reply:
I could just have the user give a dict of keywords and values, but wanted to at least do some checking to ensure they use the right values

Reply:
I suppose they could, but that wasn't the point of this for now. We could add that later, or you just make it a config option of Tool, and it's used when building this dictionary.  So far it inherits only from traitlets.HasTraits, not traitlets.config, so doesn't have configuration functionality. I even considered not using traitlets at all, but they are convenient.